src/test/java/net/pms/util/RationalTest.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Universal Media Server, for streaming any media to DLNA
+ * compatible renderers based on the http://www.ps3mediaserver.org.
+ * Copyright (C) 2012 UMS developers.
+ *
+ * This program is a free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2
+ * of the License only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package net.pms.util;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import static org.junit.Assert.*;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class RationalTest {
+
+	@Rule
+	public final ExpectedException exception = ExpectedException.none();
+
+	@Test
+	public void testDefaultConstructor() {
+		assertEquals(Rational.ZERO, new Rational());
+		assertEquals(BigInteger.ZERO, Rational.ZERO.numerator);
+		assertEquals(BigInteger.ONE, Rational.ZERO.denominator);
+		assertEquals(BigInteger.ONE, Rational.ZERO.greatestCommonDivisor);
+		assertEquals(BigInteger.ZERO, Rational.ZERO.reducedNumerator);
+		assertEquals(BigInteger.ONE, Rational.ZERO.reducedDenominator);
+		assertEquals("0", Rational.ZERO.stringValue);
+		assertEquals("0", Rational.ZERO.reducedStringValue);
+		assertEquals("0", Rational.ZERO.decimalStringValue);
+
+	}
+
+	@Test
+	public void testDoubleConstructor() {
+		assertEquals(0.342, new Rational(0.342).doubleValue(), 0.0);
+		assertEquals(10000d, new Rational(10000d).doubleValue(), 0.0);
+		assertEquals(1d/3, new Rational(1d/3).doubleValue(), 0.0);
+	}
+
+	@Test
+	public void testBigDecimalConstructor() {
+		assertEquals(0, new Rational(BigDecimal.ZERO).intValue());
+		assertEquals(BigDecimal.ONE, new Rational(BigDecimal.ONE).bigDecimalValue());
+		assertEquals(BigDecimal.valueOf(Long.MAX_VALUE), new Rational(BigDecimal.valueOf(Long.MAX_VALUE)).bigDecimalValue());
+		assertEquals(BigDecimal.valueOf(Long.MIN_VALUE), new Rational(BigDecimal.valueOf(Long.MIN_VALUE)).bigDecimalValue());
+		assertEquals(
+			new BigDecimal("398427545234132213.4398023478543978313289043509400000000000000000000000000000000000000000000000000000000000000000000000"),
+			new Rational(new BigDecimal("398427545234132213.43980234785439783132890435094")).bigDecimalValue()
```

Comment:
@Sami32 What do you mean? The tests are simply testing that it can handle big numbers, the ```Rational``` class can handle numbers with as many digits as your memory allows, so these numbers are actually very small.